site/functions/server/main.go

```diff
@@ -0,0 +1,287 @@
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		writeJSON(w, PluginCountResponse{Error: ErrorResponse{Message: fmt.Sprintf("error retrieving repo contents: %v", err)}})
+		return
+	}
+	yamls := filterYAMLs(dir)
+	count := len(yamls)
+	log.Printf("github response=%s count=%d rate: limit=%d remaining=%d",
+		resp.Status, count, resp.Rate.Limit, resp.Rate.Remaining)
+
+	var out PluginCountResponse
+	out.Data.Count = count
+
+	w.Header().Set("Cache-Control", fmt.Sprintf("public, max-age=%d", cacheSeconds))
+	writeJSON(w, out)
+}
+
+func writeJSON(w io.Writer, v interface{}) {
+	e := json.NewEncoder(w)
+	e.SetIndent("", "  ")
+	if err := e.Encode(v); err != nil {
+		log.Printf("json write error: %v", err)
+	}
+}
+
+func filterYAMLs(entries []*github.RepositoryContent) []*github.RepositoryContent {
+	var out []*github.RepositoryContent
+	for _, v := range entries {
+		if v == nil {
+			continue
+		}
+		if v.GetType() == "file" && strings.HasSuffix(v.GetName(), constants.ManifestExtension) {
+			out = append(out, v)
+		}
+	}
+	return out
+}
+
+func loggingHandler(f http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		start := time.Now()
+		log.Printf("[req]  > method=%s path=%s", req.Method, req.URL)
+		defer func() {
+			log.Printf("[resp] < method=%s path=%s took=%v", req.Method, req.URL, time.Since(start))
+		}()
+		f.ServeHTTP(w, req)
+	})
+}
+
+func pluginsHandler(w http.ResponseWriter, req *http.Request) {
+	_, dir, resp, err := githubClient(req.Context()).
+		Repositories.GetContents(req.Context(), orgName, repoName, pluginsDir, &github.RepositoryContentGetOptions{})
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		writeJSON(w, PluginsResponse{Error: ErrorResponse{Message: fmt.Sprintf("error retrieving repo contents: %v", err)}})
+		return
+	}
+	log.Printf("github response=%s rate: limit=%d remaining=%d",
+		resp.Status, resp.Rate.Limit, resp.Rate.Remaining)
+	var out PluginsResponse
+
+	plugins, err := fetchPlugins(filterYAMLs(dir))
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		writeJSON(w, PluginsResponse{Error: ErrorResponse{Message: fmt.Sprintf("failed to fetch plugins: %v", err)}})
+		return
+	}
+
+	for _, v := range plugins {
+		pi := pluginInfo{
+			Name:             v.Name,
+			Homepage:         v.Spec.Homepage,
+			ShortDescription: v.Spec.ShortDescription,
+			GithubRepo:       findRepo(v.Spec.Homepage),
+		}
+		out.Data.Plugins = append(out.Data.Plugins, pi)
+	}
+
+	w.Header().Set("Cache-Control", fmt.Sprintf("public, max-age=%d", cacheSeconds))
+	writeJSON(w, out)
+}
+
+func fetchPlugins(entries []*github.RepositoryContent) ([]*krew.Plugin, error) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var (
+		mu     sync.Mutex
+		out    []*krew.Plugin
+		retErr error
+	)
+
+	queue := make(chan string)
+	var wg sync.WaitGroup
+
+	for i := 0; i < urlFetchBatchSize; i++ {
```

Comment:
nit: urlFetchBatchSize -> pluginFetchParallelism
```suggestion
	for i := 0; i < pluginFetchParallelism; i++ {
```